sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -392,7 +376,18 @@ export class MessageSender extends LinkEntity {
   /**
    * Initializes the sender session on the connection.
    */
-  public async open(options?: AwaitableSenderOptions): Promise<void> {
+  public async open(
+    options?: AwaitableSenderOptions,
+    abortSignal?: AbortSignalLike
+  ): Promise<void> {
+    const checkAborted = (): void => {
+      if (abortSignal?.aborted) {
+        throw new AbortError("Sender creation was cancelled by the user.");
+      }
+    };
+
+    checkAborted();
```

Comment:
When sender is already open, `open()` is essentially a no-op. We can skip making this check and do an early exit right?

Reply:
This is basically the only spot the abortSignal gets checked in some code paths. In particular, createBatch() which basically just does this call and then grabs a property from the sender. 

If/when we move this logic out of here it makes sense to not have too much interdependence between how this function behaves and where it was called from.

Reply:
I get what @ramya-rao-a is saying, and I think she has a point because in this situation we're talking about aborting something that's already been completed (in the case that this.isOpen() is true), so since it's essentially a no-op, it makes sense that we wouldn't abort. Now, whatever called `open()` in the first place could check the abortSignal and determine whether it should abort or not.

That said, I could see an argument for throwing the AbortError after doing the `this.isOpen()` check but before acquiring the lock. One could say that the caller of `open` should check the abortSignal first, but I'm not sure if that's always possible from a practical standpoint.

Reply:
You both have good points. :)

I think the issue I'm having is what the contract for open() is. Is it:
a. I only abort if I'm doing network IO.
b. I abort regardless of whether or not I do network IO (ie, IO is an implementation detail)

For most clients this isn't an issue - they don't cache any data so every call that does IO _always_ does IO. That's different here since we store off the connection state and can do an early exit.

@bterlson, do you mind weighing in on this to clarify the guideline?

Reply:
I guess it doesnt make much of a difference to the end user experience either way, so am fine with the current state

Reply:
I'd definitely like to get clarification on the guideline for this case, but maybe since this isn't a public API and you're looking at eventually refactoring it anyway I'm being too pedantic.

Reply:
For this PR I can definiteliy leave it but we should get clarification - I think the AMQP-based packages are going to have issues like this where we aren't stateless so this kind of skipped-IO pattern can show up in quite a few spots.

Reply:
Just closing the loop, but after talking to @bterlson offline it sounds like we should be throwing an `AbortError` rather than exiting early, so I withdraw my argument.